tests/IntegrationTests/WebAppUiTests/AutomatedUiTests.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.Identity.Web.Test.Common;
+using Microsoft.Identity.Web.Test.LabInfrastructure;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+using Xunit;
+
+namespace WebAppUiTests
+{
+#if !FROM_GITHUB_ACTION
+    public class AutomatedUiTests
+    {
+        [Fact]
+        public async Task ChallengeUser_SignInSucceedsTestAsync()
+        {
+            // Arrange
+            LabResponse labResponse = await LabUserHelper.GetDefaultUserAsync().ConfigureAwait(false);
+
+            ChromeOptions options = new ChromeOptions();
+            // ~2x faster, no visual rendering
+            // comment-out below when debugging to see the UI automation
+            options.AddArguments(TestConstants.Headless);
+            IWebDriver driver = new ChromeDriver(options);                  
+            driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(10);            
```

Comment:
this is an implicit wait, so if the elements are taking longer to appear, we won't move to the next element and throw an error. we can adjust this, right now, it's pretty fast.